### DISCUSSION TITLE
How to submit form after every steps completed in wizard

### DISCUSSION THREAD

**Discussion by Tjajeer** [data-driven-forms/react-forms#1173] [Ideas]
component Wizard
scope:
user should be able to submit the form after every step to the backend API in the form wizard

**Comment by Hyperkid123** [comment]
@Tjajeer what component mapper do you use?

**Comment by rvsia** [comment]
@Tjajeer You have two options:

1. Each wizard step is an individual instance of the form renderer:

```jsx
<Wizard>
   <WizardStep><FormRenderer ... /></WizardStep>
   <WizardStep><FormRenderer ... /></WizardStep>
   <WizardStep><FormRenderer ... /></WizardStep>
   <WizardStep><FormRenderer ... /></WizardStep>
</Wizard>
```

2. You can customize `Buttons` to enable custom behavior in some of the mappers, for example in [PF4 mapper](https://data-driven-forms.org/provided-mappers/wizard?mapper=pf4#buttons).
